src/account/default.ts

```diff
@@ -209,6 +260,67 @@ export class Account extends Provider implements AccountInterface {
     );
   }
 
+  public async deployAccount(
+    {
+      classHash,
+      constructorCalldata = [],
+      addressSalt = 0,
+      contractAddress: providedContractAddress,
+    }: DeployAccountContractPayload,
+    transactionsDetail: InvocationsDetails = {}
+  ): Promise<DeployContractResponse> {
+    const nonce = toBN(transactionsDetail.nonce ?? (await this.getNonce()));
+    let maxFee: BigNumberish = '0';
+
+    if (transactionsDetail.maxFee || transactionsDetail.maxFee === 0) {
+      maxFee = transactionsDetail.maxFee;
+    } else {
+      const { suggestedMaxFee } = await this.estimateAccountDeployFee(
+        { addressSalt, classHash, constructorCalldata },
+        {
+          nonce,
+        }
+      );
+      maxFee = suggestedMaxFee.toString();
+    }
+
+    const version = toBN(transactionVersion);
```

Comment:
I would maybe move these `version`, `chainId ` and `contractAddress ` to the start of the method to have it consistent with `estimateAccountDeployFee`